client/app/scripts/utils/update-buffer-utils.js

```diff
@@ -0,0 +1,129 @@
+import _ from 'lodash';
+import debug from 'debug';
+import Immutable from 'immutable';
+
+import { receiveNodesDelta, receiveNodeDetails } from '../actions/app-actions';
+import AppStore from '../stores/app-store';
+
+const log = debug('scope:update-buffer-utils');
+const makeList = Immutable.List;
+const feedInterval = 1000;
+const bufferLength = 100;
+
+let deltaBuffer = makeList();
+let detailsBuffer = makeList();
+let updateTimer = null;
```

Comment:
Again single immutable store would be awesome here! Then could pin the whole UI to some ref of that while the `HEAD` keeps moving forward w/ updates.